src/core/net/srp_server.cpp

```diff
@@ -438,15 +438,32 @@ void Server::CommitSrpUpdate(Error                    aError,
 
 void Server::Start(void)
 {
-    Error error = kErrorNone;
+    Error    error = kErrorNone;
+    uint16_t port  = kUdpPortMin;
 
-    VerifyOrExit(!IsRunning());
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE && OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO
+    Settings::SrpServerInfo info;
+    if (Get<Settings>().Read(info) == kErrorNone)
+    {
+        port = info.GetPort() + 1;
+        if (port > kUdpPortMax)
```

Comment:
What if port < kUdpPortMin?

Reply:
👍  very good point @simonlingoogle. We need to handle such a case (i.e., the config min/max port range may be different between version that saved the value vs the one reading it later).